File: src/main/java/com/ndgndg91/ordermatchingengine/order/dto/request/AddOrderRequest.java
package com.ndgndg91.ordermatchingengine.order.dto.request;

import com.ndgndg91.ordermatchingengine.order.OrderType;
import com.ndgndg91.ordermatchingengine.order.PriceType;
import com.ndgndg91.ordermatchingengine.order.Symbol;
import lombok.Getter;
import lombok.ToString;

import java.math.BigDecimal;

@Getter
@ToString
public final class AddOrderRequest {
    private OrderType orderType;
    private Symbol symbol;
    private int shares;
    private PriceType priceType;
    private BigDecimal price;
}

File: OrderMatchingEngineKotlin/src/main/kotlin/com/ndgndg91/ordermatchingenginekotlin/order/dto/request/AddOrderRequest.kt
package com.ndgndg91.ordermatchingenginekotlin.order.dto.request

import java.math.BigDecimal

data class AddOrderRequest(
    val orderType: String,
    val symbol: String,
    val shares: Int,
    val priceType: String,
    val price: BigDecimal
)
